### PR TITLE
mgmt: hawkbit: change the tls certificate tag 

### PR DESCRIPTION
**Pull request by maass-hamburg**
Be able to change the tls certificate tag in hawkBit 
and be able to deactivate TLS for the hawkBit subsystem independent of `CONFIG_NET_SOCKETS_SOCKOPT_TLS`

### CODE REVIEW

**Comment by maass-hamburg** [comment]
@real-tintin

**Review by real-tintin — APPROVED** [review]

File: subsys/mgmt/hawkbit/hawkbit.c [hunk 1]
```diff
@@ -60,17 +60,13 @@ static bool hawkbit_initialized;
 
 #ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
 
-#ifdef CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY
-#define AUTH_HEADER_START "Authorization: GatewayToken "
-#else
-#define AUTH_HEADER_START "Authorization: TargetToken "
-#endif /* CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY */
+#define AUTH_HEADER_START                                                                          \
```
Comment (ycsin) [inline subsys/mgmt/hawkbit/hawkbit.c:58]: We can probably use some Kconfig magic here instead:

```kconfig
config HAWKBIT_DDI_SECURITY_HEADER_START
    string
    default "Authorization: GatewayToken " if HAWKBIT_DDI_GATEWAY_SECURITY
    default "Authorization: TargetToken " if HAWKBIT_DDI_TARGET_SECURITY
    help
      Internal helper Kconfig blah blah blah
```

File: subsys/mgmt/hawkbit/hawkbit.c [hunk 2]
```diff
@@ -60,17 +60,13 @@ static bool hawkbit_initialized;
...
+	"Authorization: " COND_CODE_1(CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY, ("Gateway"),            \
+				      ("Target")) "Token "
 
-#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
-#define AUTH_HEADER_FULL AUTH_HEADER_START "%s" HTTP_CRLF
-#else
-#define AUTH_HEADER_FULL AUTH_HEADER_START CONFIG_HAWKBIT_DDI_SECURITY_TOKEN HTTP_CRLF
-#endif /* CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME */
+#define AUTH_HEADER_FULL                                                                           \
```
Comment (ycsin) [inline subsys/mgmt/hawkbit/hawkbit.c:62]: Yes, it is shorter now, but I'm not sure if this made it easier to read. I'd rather have a few more lines in the source file than something that's a few lines shorter but harder to read..

File: include/zephyr/mgmt/hawkbit.h [hunk 3]
```diff
@@ -154,7 +157,7 @@ struct hawkbit_runtime_config hawkbit_get_config(void);
 static inline int hawkbit_set_server_addr(char *addr_str)
 {
 	struct hawkbit_runtime_config set_config = {
-		.server_addr = addr_str, .server_port = 0, .auth_token = NULL};
+		.server_addr = addr_str, .server_port = 0, .auth_token = NULL, .tls_tag = 0};
```
Comment (ycsin) [inline include/zephyr/mgmt/hawkbit.h:160]: Hmm.. imagine that if we do not provide these functions, we probably don't need to modify this file at all

File: subsys/mgmt/hawkbit/hawkbit.c [hunk 4]
```diff
@@ -75,9 +70,9 @@ static struct hawkbit_config {
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 	char server_addr[DNS_MAX_NAME_SIZE + 1];
 	char server_port[sizeof(STRINGIFY(__UINT16_MAX__))];
-#ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
-	char ddi_security_token[DDI_SECURITY_TOKEN_SIZE + 1];
-#endif
+	IF_DISABLED(CONFIG_HAWKBIT_DDI_NO_SECURITY,
+		    (char ddi_security_token[DDI_SECURITY_TOKEN_SIZE + 1];))
+	IF_ENABLED(CONFIG_HAWKBIT_USE_DYNAMIC_CERT_TAG, (sec_tag_t tls_tag;))
```
Comment (ycsin) [inline subsys/mgmt/hawkbit/hawkbit.c:75]: IMHO the old implementation is more readable and generally accepted within Zephyr for struct declaration. The `IF_ENABLED`/`IF_DISABLED` is generally used inline in C code or in macros for initialization

File: subsys/mgmt/hawkbit/hawkbit.c (hunk 1 above)
Comment (maass-hamburg) [reply subsys/mgmt/hawkbit/hawkbit.c]: For the first we will leave it like it is.

File: include/zephyr/mgmt/hawkbit.h (hunk 3 above)
Comment (maass-hamburg) [reply include/zephyr/mgmt/hawkbit.h:160]: Well now we have them. and if it's getting to long we could still change it to 
```
struct hawkbit_runtime_config set_config = {0};
set_config.server_addr = addr_str;
```

**Review by ycsin — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**